cmd/kubectl-k8ssandra/register/registration.go

```diff
@@ -0,0 +1,143 @@
+package register
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8ssandra/k8ssandra-client/pkg/registration"
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
```

Comment:
issue: Do not include pkg from k8ssandra-operator as we want to avoid circular dependencies. This project is intended to be used by the cass-operator / k8ssandra-operator / other operators / software.